scripts/create_dataset.py

```diff
@@ -46,26 +46,323 @@ def init_dataset(dataset_name, dataset_description):
+    asset_with_tables_and_follow = add_asset_follow(asset=asset_with_tables, dataset=dataset)
+
+    return asset_with_tables_and_follow
+
+
+def add_asset_tables(asset, dataset):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset: the asset to be created and added to the dataset with table names
+    :param dataset: the created dataset that will be written to disk
+    """
+    for table in dataset["schema"]["tables"]:
+        asset["tables"].append(
+            {
+                "name": table["name"],
+                "columns": []
+            }
+        )
+    return asset
+
+
+def add_asset_follow(asset, dataset):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset: the asset to be created and added to the dataset with relationship names
+    :param dataset: the created dataset that will be written to disk
+    """
+    for relationship in dataset["schema"]["relationships"]:
+        asset["follow"].append(relationship["name"])
+
+    return asset
+
+
+def init_asset(asset_name, root_table, root_column):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset_name: the name of the asset being  created and added to the dataset
+    :param root_table: the name of the root table for the asset
+    :param root_column: the name of the root column for the asset
+    """
+    asset = {
+        "name": asset_name,
+        "rootTable": root_table,
+        "rootColumn": root_column,
+        "tables": [],
+        "follow": []
+    }
+    return asset
+
+
+def create_table_and_relationships(input_table, input_tables):
+    """
+    Creates a table and relationships using the input tables fields and reference field.
+    :param input_table: the input table to be used to create a new table and relationships
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    table_name = input_table["name"]
+    table = init_table(table_name=table_name)
+    relationships = []
+    for input_column in input_table["columns"]:
+        column, new_relationships = create_column_and_relationships(input_column=input_column,
+                                                                    table_name=table_name,
+                                                                    input_tables=input_tables)
+        table["columns"].append(column)
+
+        # only add a relationship to the table if the given input column contains reference(s)
+        if len(new_relationships) > 0:
+            relationships.extend(new_relationships)
+
+    return table, relationships
+
+
+def init_table(table_name):
+    """
+    Creates a new table as dict with a table name and an empty list for the columns field.
+    :param table_name: the name of input table to be used to create a new table
+    """
+    table = {
+        "name": table_name,
+        "columns": []
+    }
+
+    return table
```

Comment:
Same as with `init_asset`, I don't think it's worth having a separate method for this.